sha2/src/lib.rs

```diff
@@ -13,23 +13,37 @@
 #[cfg(not(any(target_arch = "x86_64", target_arch = "x86", target_arch = "aarch64")))]
 compile_error!("crate can only be used on x86, x86-64 and aarch64 architectures");
 
+cpufeatures::new!(cpuid_avx2, "avx2");
```

Comment:
Gate this line on `#[cfg(any(target_arch = "x86_64", target_arch = "x86"))]`. Otherwise it causes compilation failure on Aarch64 targets.

Reply:
You forgot to modify the `compress256` function (see the CI failure). Currently it tries to use the `cpuid_avx2` module on all targets. I think the easiest solution would be to introduce two function with the same name one gated on x86(-64) and another one on AArch64.